daffodil-core/src/main/scala/org/apache/daffodil/dpath/DFDLExpressionParser.scala

```diff
@@ -157,7 +157,14 @@ class DFDLPathExpressionParser[T <: AnyRef](qn: NamedQName,
           nextRdr = nextRdr.rest
           i += 1
         }
-        context.SDE("Unable to parse expression. Message: %s\nNext: %s.", msg, nextString.toString())
+        val err_msg = {
+          if (msg.contains('/')) {
+            msg + ", did you mean to use 'div'?"
+          } else {
```

Comment:
This feels pretty fragile to me, and could potentially give a lot of false negatives if there's some error message that uses a /, like maybe a path expression. Perhaps an alternative approach would be to modify the MultiplicativeExpr grammar to accept a /, but then call context.SDE() if it matches? We do similar things for `//`` in a path expression. Not sure if there might be some ambiguities and adding that might make normal paths (i.e. ``/foo/bar/baz``) look like errors in some cases, though.

Reply:
While I agree that that is a better solution, it seems to break all the path tests where as my original method doesn't. I'm not entirely sure I understand the code for the UnaryExpr stuff correctly, so it's possible I did something wrong. I'll push up my changes as an example.